pkg/apis/serving/v1beta1/revision_lifecycle.go

```diff
@@ -26,6 +27,13 @@ const (
 )
 
 // GetGroupVersionKind returns the GroupVersionKind.
-func (r *Revision) GetGroupVersionKind() schema.GroupVersionKind {
+func (*Revision) GetGroupVersionKind() schema.GroupVersionKind {
 	return SchemeGroupVersion.WithKind("Revision")
 }
+
+// IsReady returns if the revision is ready to serve the requested revision
+// and the revision resource has been observed.
+func (r *Revision) IsReady() bool {
+	rs := r.Status
+	return rs.ObservedGeneration == r.Generation && apis.NewLivingConditionSet().Manage(&rs).IsHappy()
```

Comment:
For this to exhibit the same behaviour as v1 & alpha APIs you need to build the same condition set.